### DISCUSSION TITLE
I want to send a welcome message to the player. Is there an event for when the player leaves CONFIGURATION state and enters PLAY state?

### DISCUSSION THREAD

**Discussion by EnbyAce** [Minestom/Minestom#2440] [Help] [answered]
I want to send a welcome message to the player when they join, but I'm getting an error saying `java.lang.IllegalStateException: Packet SystemChatPacket is not valid in state CONFIGURATION (only PLAY)`.
How do I fix it?

Currently trying to do it as an `AsyncPlayerConfigurationEvent` because I don't know which other event I should use.

**Comment by cosrnic** [comment, marked as answer]
use `PlayerSpawnEvent`, if you need it to only activate on first spawn, you can check `event.isFirstSpawn()`

**Reply by EnbyAce** [reply]
Thanks, I'll give that a go.
